### PR TITLE
Resource management API cleanup

### PR DESCRIPTION
**Pull request by alexeykazakov**
- Add RoleManagementModelService to application
- Create RoleManagementModelService via application interface
- Remove Role Management Controller service (use RoleManagementModelService directly from controller)
- Mount Roles controller
- Remove Repositories interface

### CODE REVIEW

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-454-1`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-454-2`

File: controller/resource_roles.go [hunk 1]
```diff
@@ -33,15 +30,28 @@ func (c *ResourceRolesController) ListAssigned(ctx *app.ListAssignedResourceRole
 
 	var roles []role.IdentityRole
 
-	roles, err := c.roleManagementService.ListByResource(ctx, ctx.ResourceID)
+	err := application.Transactional(c.db, func(appl application.Application) error {
```
Comment (sbose78) [inline controller/resource_roles.go]: This is where the new design shows up - basically the controller calling the 'model' and repository.

File: controller/resource_roles.go [hunk 2]
```diff
@@ -69,14 +93,15 @@ func (c *ResourceRolesController) ListAssignedByRoleName(ctx *app.ListAssignedBy
...
 	var rolesList []*app.IdentityRolesData
 	for _, r := range roles {
-		rolesList = append(rolesList, convertIdentityRoleToAppRole(ctx, r))
+		rolesList = append(rolesList, convertIdentityRoleToAppRole(r))
 	}
 	return rolesList
 }
-func convertIdentityRoleToAppRole(ctx context.Context, r role.IdentityRole) *app.IdentityRolesData {
+
+func convertIdentityRoleToAppRole(r role.IdentityRole) *app.IdentityRolesData {
```
Comment (sbose78) [inline controller/resource_roles.go]: Good cleanup - we 'overuse' the passing of the context variable :(

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-454-3`

**Comment by sbose78** [comment]
[test]

File: authorization/permission/model/permission_model_service.go [hunk 3]
```diff
@@ -13,14 +13,13 @@ type PermissionService interface {
 
 // GormPermissionModelService is the implementation of the interface for
 // PermissionModelService. IMPORTANT NOTE: Transaction control is not provided by this service
-type PermissionServiceImpl struct {
-	PermissionService
+type GormPermissionModelService struct {
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply authorization/permission/model/permission_model_service.go]: This is actually a step in the wrong direction, as it is further entwining the service layer with an implementation detail.  Can we please hold off on this PR until we discuss our architecture further.

[the remark this answers is not in the document]
Comment (sbose78) [reply authorization/permission/model/permission_model_service.go]: But this layer is always supposed to talk to the database related objects ?

[the remark this answers is not in the document]
Comment (alexeykazakov) [reply authorization/permission/model/permission_model_service.go]: This PR fixes a few bugs and doesn’t change our existing arch.
I’m going to merge it. Let’s address arch concerns in a separate PR.

**Pull request merged**